### ISSUE TITLE
When using Test template formatting appears  incorrect

### ISSUE THREAD

**Issue by GerwinLaagland** [nokia/RED#297]
When using a 'Test Template' settings option, the presentation indicates the first value to be a keyword when it is not. A minor cosmetic issue, though it does appear both in Source and Testcase views. The test runs perfectly fine, so it seems to be only an issue with the presentation.

No further validations are triggered either, so could also be that I'm doing something wrong ofcourse.

Versions used:
Eclipse: 2019-03
Red: 0.8.11

Example testcase:
```
*** Settings ***
Test Template    kw

*** Keywords ***
kw    [Arguments]    ${USER}    ${PASSWORD}    
    Log    argument1 = ${USER}
    Log    argument2 = ${PASSWORD}

*** Test Cases ***      
Testcase
    #USERNAME    PASSWORD
     valid       valid 
     invalid     valid
     valid       invalid
     ${EMPTY}    valid
     valid       ${EMPTY}
     ${EMPTY}    ${EMPTY}    
```

Screenshot of the presentation in source view, note the blue values:
![image](https://user-images.githubusercontent.com/8427040/56690361-8f548f80-66dd-11e9-9f4a-24036e6c9fc6.png)

Screenshot of the presentation in Test case view, note again the blue values:
![image](https://user-images.githubusercontent.com/8427040/56690409-ae532180-66dd-11e9-9659-b683215b8927.png)

My expectation would be that the blue colour would the 'normal' black variety seen in the second column of the first three testcases.

Reference of the notation option in RF documentation:
http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-templates

**Comment by michalanglart** [comment]
Hi,

no - you're not doing anything wrong. This is still not implemented (both syntax coloring and validation). We have it in our backlog and it even used to be near the top of it however more important issues were popping up so it is still not started.

Additionally it seems that templates are not heavily used since this is the first time this issue is reported here at GitHub :). We'll make sure that those issues will get to the top of the backlog eventually.

**Comment by GerwinLaagland** [comment]
Good news! I agree with the sentiment that it is a functionality that is not used quite often. I was researching some notation options that Robot Framework provided, and someone mentioned this to me. Otherwise I wouldn't have known still.

It certainly hasn't seen any use in my projects till date. So it is no more than logical that it is low on the priority list. :-) Thanks for your continued efforts!

**Comment by humbienri** [comment]
I actually do use Templates in Robot Framework and noticed this "coloration" anomaly but since the code/tests work as expected I didn't think to actually report the buggy behaviour.  I do agree it's not urgent and more on the nice to have list.  But I fear that will always be the case and thus this will never truly rise to the top also.

Thank you fellas!

**Comment by michalanglart** [comment]
Templates are now fully supported with colouring, validation and content assistant being aware of them. Implemented in 0.9.0
